packages/realm/src/Object.ts

```diff
@@ -185,7 +185,11 @@ export class RealmObject<T = DefaultObject> {
     const table = binding.Helpers.getTable(realm.internal, tableKey);
     if (primaryKey) {
       const primaryKeyHelpers = properties.get(primaryKey);
-      const primaryKeyValue = values[primaryKey];
+      const defaultValue = primaryKeyHelpers.default;
+      let primaryKeyValue = values[primaryKey];
+      if (primaryKeyValue === undefined && typeof defaultValue !== "undefined") {
```

Comment:
A few things:

1. we shouldn't waste time looking up `primaryKeyHelpers.default` unless `primaryKeyValue` is undefined.
2. why mix `=== undefined` with `typeof x !== "undefined"` style checks. We should probably have a rule about which is prefered and stick to it. (I propose `=== undefined` unless specially looking for a global variable, but even then I might prefer `globalThis.blah === undefined` for explicitness).
3. I think you can just remove `&& typeof defaultValue !== "undefined"` entirely without affecting the behavior of this code.

Reply:
Good point.  I've simplified the logic to be a more straightforward